src/ansible_navigator/actions/lint.py

```diff
@@ -0,0 +1,372 @@
+Internally, it works by using ansible-runner to execute ansible-lint (optionally
+in an execution environment). When running ansible-lint, it passes
+'-f codeclimate' which requests JSON output on stdout. The JSON output contains
+a list of issues which we then report in the ansible-navigator UI.
+
+We allow users to dig into an individual issue using the standard :<lineno>
+commands, to learn more about the issue.
+
+The full specification for ansible-lint's use of JSON (for the codeclimate
+formatter) can be found in src/ansiblelint/formatters/__init__.py in the
+ansible-lint codebase.
+"""
+
+import collections
+import curses
+from dataclasses import asdict, dataclass
+from enum import IntEnum
+from functools import total_ordering
+import json
+import os
+import shutil
+import shlex
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+from . import _actions as actions
+from . import run_action
+from ..app import App
+from ..app_public import AppPublic
+from ..configuration_subsystem import ApplicationConfiguration
+from ..runner.api import CommandRunner
+from ..steps import Step
+from ..ui_framework import (
+    CursesLines,
+    CursesLinePart,
+    Interaction,
+    nonblocking_notification,
+    error_notification,
+)
+from ..utils import abs_user_path
+
+
+class Severity(IntEnum):
+    info = 10
+    minor = 20
+    major = 30
+    critical = 40
+    blocker = 50
+
+    # If ansible-lint ever gives us something we don't expect, return this.
+    # In practice, we reverse-sort, so this will always be last.
+    unknown = -1
+
+    @classmethod
+    def _missing_(cls, value):
+        """
+        If ansible-lint ever gives us something we don't expect return something
+        that tells us.
+        """
+        return Severity.unknown
+
+    # ui_framework.menu_builder.MenuBuilder will str() whatever it is given.
+    def __str__(self):
+        if self == Severity.unknown:
+            return "<unknown>"
+        return self.name
+
+
+@dataclass
+@total_ordering
+class LintResult:
+    """A single result, parsed from Ansible Lint."""
+
+    raw: dict
+    message: str
+    severity: Severity
+    line: int
+    column: Optional[int]
+    path: str
+    description: str
+
+    def __lt__(self, other):
+        """Allow for easy sorting of results by severity."""
+        return self.severity < other.severity
+
+    @staticmethod
+    def from_raw(issue: Dict[str, Any]):
+        """
+        Massages the raw JSON output into our LintResult structure and returns a
+        a list of them.
+        """
+
+        column = None
+        # This will be a dict if we have a line + column, otherwise an int
+        # with just the line number.
+        if isinstance(issue["location"]["lines"]["begin"], collections.Mapping):
+            line = issue["location"]["lines"]["begin"]["line"]
+            column = issue["location"]["lines"]["begin"]["column"]
+        else:
+            line = issue["location"]["lines"]["begin"]
```

Comment:
I feel like this can be simplified by reducing copy-paste when accessing nested locations and labeling things:
```suggestion
        beginning_marker = issue["location"]["lines"]["begin"]

        line = beginning_marker
        column = None

        # This will be a dict if we have a line + column, otherwise an int
        # with just the line number.
        is_a_2d_coordinate = isinstance(beginning_marker, collections.Mapping)

        if is_a_2d_coordinate:
            line = beginning_marker["line"]
            column = beginning_marker["column"]
```